### PR TITLE
TMA-366: Support technical_users (plural) param

### PR DESCRIPTION
**Pull request by panjan**
RELATED: TMA-366

### CODE REVIEW

File: spec/unit/actions/ensure_technical_users_domain_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  let(:user) { double(:user) }
+  before do
+    allow(gdc_gd_client).to receive(:domain) { domain }
+    allow(domain).to receive(:users) { [] }
+    allow(user).to receive(:login)
+    allow(user).to receive(:email)
+    allow(domain).to receive(:add_user) { user }
+  end
+
+  context 'when deprecated param technical_user specified' do
```
Comment (yenkins) [inline spec/unit/actions/ensure_technical_users_domain_spec.rb]: Similar code found in ensure_technical_users_domain_spec.rb:29, ensure_technical_users_domain_spec.rb:40 (mass = 42)

File: spec/unit/actions/ensure_technical_users_domain_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      params = {
+        gdc_gd_client: gdc_gd_client,
+        technical_user: ['foo@bar.com']
+      }
+      GoodData::LCM2.convert_to_smart_hash(params)
+    end
+    it_behaves_like 'a technical users domain action'
+  end
+
+  context 'when param technical_users specified' do
```
Comment (yenkins) [inline spec/unit/actions/ensure_technical_users_domain_spec.rb]: Similar code found in ensure_technical_users_domain_spec.rb:29, ensure_technical_users_domain_spec.rb:40 (mass = 42)

File: spec/unit/actions/ensure_technical_users_project_spec.rb [hunk 3]
```diff
@@ -7,43 +7,66 @@
...
 
-    before do
-      allow(project).to receive(:title) { 'foo project' }
-      allow(project).to receive(:pid) { 'abcdefg' }
-      allow(project).to receive(:create_users) { [{ type: 'successful', message: 'yahoo!', user: '/gdc/account/foo' }] }
-      allow(gdc_gd_client).to receive(:projects) { project }
+    it_behaves_like 'a technical users project action'
+  end
+
+  context 'when deprecated param technical_user specified' do
```
Comment (yenkins) [inline spec/unit/actions/ensure_technical_users_project_spec.rb]: Similar code found in ensure_technical_users_project_spec.rb:49, ensure_technical_users_project_spec.rb:61 (mass = 58)

File: spec/unit/actions/ensure_technical_users_project_spec.rb [hunk 4]
```diff
@@ -7,43 +7,66 @@
...
-      result = subject.class.call(params)
-      expected = [{ project: 'foo project',
-                    pid: 'abcdefg',
-                    login: 'foo@bar.com',
-                    role: 'admin',
-                    result: 'successful',
-                    message: 'yahoo!',
-                    url: '/gdc/account/foo' }]
-      expect(result).to eq(expected)
+  context 'when param technical_users specified' do
```
Comment (yenkins) [inline spec/unit/actions/ensure_technical_users_project_spec.rb]: Similar code found in ensure_technical_users_project_spec.rb:49, ensure_technical_users_project_spec.rb:61 (mass = 58)

File: lib/gooddata/lcm/actions/ensure_technical_users_domain.rb [hunk 5]
```diff
@@ -36,7 +39,7 @@ def call(params)
           domain_name = params.organization || params.domain
           domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
 
-          technical_users = (params.technical_user || []).uniq
+          technical_users = (params.technical_users || params.technical_user || []).uniq
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/ensure_technical_users_domain.rb]: Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/lcm/actions/ensure_technical_users_project.rb [hunk 6]
```diff
@@ -36,7 +39,7 @@ class << self
         def call(params)
           client = params.gdc_gd_client
 
-          technical_users = params.technical_user || []
+          technical_users = params.technical_users || params.technical_user || []
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/ensure_technical_users_project.rb]: Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)

File: spec/unit/actions/ensure_technical_users_domain_spec.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata/lcm/actions/ensure_technical_users_project'
+require 'gooddata/lcm/lcm2'
+
+shared_examples 'a technical users domain action' do
+  it 'adds technical user to the domain' do
+    expect(domain).to receive(:add_user).with({ login: 'foo@bar.com', email: 'foo@bar.com' })
```
Comment (yenkins) [inline spec/unit/actions/ensure_technical_users_domain_spec.rb]: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/367/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/102/ : FAILURE in 1m 58s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/412/ : SUCCESS in 1m 57s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/386/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/121/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/431/ : SUCCESS in 30s

**Comment by hacxman** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/155/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/122/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/432/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1030/ : SUCCESS in 42m 16s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/788/ : SUCCESS in 19m 54s

**Pull request merged**